frontend/app/login/components/LoginForm.tsx

```diff
@@ -2,17 +2,80 @@
 
 import { Button } from 'components/buttons/Button'
 import { Input } from 'components/forms/Input'
+import { useRouter } from 'next/navigation'
+import { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import useLogin from '../hooks/useLogin'
+
+type FormType = {
+  username: string
+  password: string
+}
 
 export const LoginForm: React.FC = () => {
-  const handleOnSubmit = () => {
-    console.log('SUBMITTED!')
+  const router = useRouter()
+  const { mutate, isLoading } = useLogin()
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isSubmitting }
+  } = useForm<FormType>()
+  const [isError, setIsError] = useState(false)
```

Comment:
What is the point of `isError` boolean flag if your html has this 

```
isError={errors.username?.message}
```

What if, instead of declaring another state, you can just make use of the existing `errors` formState? This would reduce the amount of states in this component.

IMO - I'm a bit more cautious when using state. It all comes down to - do I really need it? Can I infer the state from an existing state value holder?

Reply:
^ FYI this is open to discussion. I'm not shooting down your work :)